Src/DependencyCollector/Shared/ServiceBusDiagnosticsListener.cs

```diff
@@ -0,0 +1,172 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+    /// <summary>
+    /// Implements ServiceBus DiagnosticSource events handling.
+    /// </summary>
+    internal class ServiceBusDiagnosticsEventHandler
+    {
+        public const string DiagnosticSourceName = "Microsoft.Azure.ServiceBus";
+        private const string StatusPropertyName = "Status";
+        private const string EntityPropertyName = "Entity";
+        private const string EndpointPropertyName = "Endpoint";
+        private const string ExceptionPropertyName = "Exception";
+
+        // We want to reflect in the dependency type that it's one of the queue operations.
+        // This way, UI could have special representation for all kinds of queues
+        // Not knowing particular service
+        private const string DependencyType = "Queue.ServiceBus";
```

Comment:
how would you handle (on the Portal side) Azure Storage Queue, Event Hub, Kafka? Presumably, we want to have a special UI for the queues: e.g. show a node for the queue itself.

Would we have to maintain a list of the all possible dependency types that represent a queue? Could we allow custom queue instrumentations and clients then?

Reply:
well we already use "Azure queue" and I think the type names should be user friendly. And we cannot map display name in UX because there would be a disconnect form data in Kusto/Analytics.

Not sure what is the right answer to identifying queue-like services, but we need to figure this out for Azure queues anyway

Reply:
my point is: all queues should be shown differently than RPS/HTTP calls on the AppMap.
So we need to be able to efficiently tell from the data that it's a queue.
Only after that we may need to know if that's a SB or Storage Queue or EventHub or Kafka, etc...

And if Kafka is instrumented outside of our control, they could ensure that type is queue and it magically works on the portal. Otherwise, we would need to make sure portal knows about this implementation and would show it as a queue and not as standard RPC call.

So the question is: how do we ensure this now. We should not use customDimensions as they are inefficient to query and we should use default properties.

Agreed about user friendliness on the portal. And I understand your concern about consistency between what AppMap shows on the node. though I'm not sure there is an expectation that it maps directly to kusto data.

Anyway do you know if there was a discussion on this problem and perhaps there is some poposal?:

> my point is: all queues should be shown differently than RPS/HTTP calls on the AppMap.
So we need to be able to efficiently tell from the data that it's a queue.

I.e. what should we do to ensure it?

Reply:
I don't know about such discussion, I will start one

Reply:
Re: visualizing queues differently than HTTP calls. I think on top of it we have at least two types of queues - standard queue (Azure Queue) and multiplexer queue (EventHub). I would expect different visualization for these queues. So, if we're solving the problem how to differentiate dependencies then maybe we should think about different queue types as well.